osbugs-9229: Gracefully shutting down (3rd attempt)

nalhadef opened the pull request:
Original PR: https://github.com/openshift/openshift-docs/pull/45621

nalhadef commented:
/jira refresh

nalhadef commented:
/jira refresh

nalhadef commented:
@lyman9966, I hope everything is in place now. I've attached the shutting doown gracefully section here. Thanks!

Neal

[Shutting down a cluster gracefully _ Backup and restore _ OpenShift Container Platform Branch Build.pdf](https://github.com/openshift/openshift-docs/files/12704091/Shutting.down.a.cluster.gracefully._.Backup.and.restore._.OpenShift.Container.Platform.Branch.Build.pdf)

lyman9966 commented on modules/graceful-shutdown.adoc:
```diff
@@ -1,102 +1,90 @@
+* Node failure due to hardware
+* Network connectivity issues
+If your cluster fails to recover, follow the steps to restore to a previous cluster state.
+====
```
the content in block "====" didn't have correct format. It showed like this:
It is important to take an etcd backup before performing this  procedure so that your cluster can be restored if you encounter any  issues when restarting the cluster. For example, the following conditions can cause the restarted cluster to  malfunction: * etcd data corruption during shutdown * Node failure due to hardware * Network connectivity issues If your cluster fails to recover, follow the steps to restore to a  previous cluster state.

And "restore to a previous cluster state." in the last sentence should be attached a link jumping relevant page.

nalhadef replied:
Removed because the same content was repeated in https://github.com/openshift/openshift-docs/pull/64574/files

lyman9966 commented on modules/graceful-shutdown.adoc:
```diff
@@ -1,102 +1,90 @@
-----
-sh-5.1# openssl x509 -in /var/lib/kubelet/pki/kubelet-client-current.pem -noout -enddate
+$ for node in $(oc get nodes -o jsonpath='{.items[*].metadata.name}'); do echo ${node} ; oc adm cordon ${node} ; done
 ----
```
Please check the "Example output" of this step, it is not correct.  And the following content like "Check the kubelet server certificate expiration date by running the following command:" is not relevant and it appears twice. Please confirm and delete

lyman9966 commented on modules/graceful-shutdown.adoc:
```diff
@@ -1,102 +1,90 @@
-<1> `-h 1` indicates how long, in minutes, this process lasts before the control-plane nodes are shut down. For large-scale clusters with 10 nodes or more, set to 10 minutes or longer to make sure all the compute nodes have time to shut down first.
++
+. Shut down all of the nodes in the cluster. You can do this from your cloud provider’s web console, or run the following loop:
 +
```
this step lacked of command of shutting down nodes, like "for node in $(oc get nodes -o jsonpath='{.items[*].metadata.name}'); do oc debug node/${node} -- chroot /host shutdown -h 1; done"

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -12,91 +12,86 @@ You can shut down your cluster in a graceful manner so that it can be restarted
-[source,terminal]
-----
-$ oc debug node/<node_name>
+[source,yaml]
```
```suggestion
[source,terminal]
```

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -12,91 +12,86 @@ You can shut down your cluster in a graceful manner so that it can be restarted
-----
-
-... Check the kubelet server certificate expiration date by running the following command:
+ci-ln-mgdnf4b-72292-n547t-master-0
```
The output needs to be separated from the command.

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -12,91 +12,86 @@ You can shut down your cluster in a graceful manner so that it can be restarted
+node/ci-ln-mgdnf4b-72292-n547t-worker-b-cmc9k cordoned
+ci-ln-mgdnf4b-72292-n547t-worker-c-vcmtn
+node/ci-ln-mgdnf4b-72292-n547t-worker-c-vcmtn cordoned
+--------
```
```suggestion
----
```

Changing this line should address the current build failure.

nalhadef commented:
Incorporated Kathryn's comments, but build still failed.

kalexand-rh commented:
> Incorporated Kathryn's comments, but build still failed.

No, the build passed: https://app.travis-ci.com/github/openshift/openshift-docs/builds/266559358

nalhadef commented:
October 13 review preview:
https://docs.openshift.com/container-platform/4.14/backup_and_restore/graceful-cluster-shutdown.html

kalexand-rh commented:
> October 13 review preview: https://docs.openshift.com/container-platform/4.14/backup_and_restore/graceful-cluster-shutdown.html

Please use this preview instead: https://65094--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-shutdown.html

kalexand-rh commented:
@nalhadef, it looks like you've picked up a large number of extra files. Will you please save a copy of the file you're working on locally and remove your second commit?

nalhadef commented:
I will @kalexand-rh. Not sure how it happened as I haven't touched this PR since Friday and it was only a single commit at that time. I'll try to fix it.

nalhadef commented:
Kathryn, can we just close the PR? Michael's PR has run and removed the
content already.


On Mon, Oct 16, 2023 at 11:06 AM Kathryn Alexander ***@***.***>
wrote:

> @nalhadef <https://github.com/nalhadef>, it looks like you've picked up a
> large number of extra files. Will you please save a copy of the file you're
> working on locally and remove your second commit?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/65094#issuecomment-1764690603>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT7EEZY32S5ZB6I637XGPCDX7VEQFAVCNFSM6AAAAAA5DEFUQ6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONRUGY4TANRQGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

kalexand-rh commented:
@nalhadef, Michael's PR removed an [assembly](https://github.com/openshift/openshift-docs/pull/65842/files#diff-7c09a5ccc63f204d75ed89dd359a340a43551a4a790ae36fb9c64f6c5149211bL2333) from the node section. This file is in the [backup and restore section](https://github.com/openshift/openshift-docs/blob/main/_topic_maps/_topic_map.yml#L2750) and is still present.

nalhadef commented:
Thanks, Kathryn.

That topic you linked to was already in place before I worked on the bug.
All I did was add a row to a table. I'll create a new PR to remove the
entire graceful shutdown topic.

On Mon, Oct 16, 2023 at 1:19 PM Kathryn Alexander ***@***.***>
wrote:

> @nalhadef <https://github.com/nalhadef>, Michael's PR removed an assembly
> <https://github.com/openshift/openshift-docs/pull/65842/files#diff-7c09a5ccc63f204d75ed89dd359a340a43551a4a790ae36fb9c64f6c5149211bL2333>
> from the node section. This file is in the backup and restore section
> <https://github.com/openshift/openshift-docs/blob/main/_topic_maps/_topic_map.yml#L2750>
> and is still present.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/65094#issuecomment-1764930158>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT7EEZ6L2VIYNA2GDMX7ZMDX7VUA3AVCNFSM6AAAAAA5DEFUQ6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONRUHEZTAMJVHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

kalexand-rh commented:
@nalhadef, this part of the backup and restore docs isn't related to the node feature that Michael removed. You can open a new PR if you are having trouble fixing this one, but this assembly needs to remain. It's been here since 4.5, and it's just an odd coincidence that they have similar names.

nalhadef commented:
OK, thanks, Kathryn.

On Mon, Oct 16, 2023 at 3:06 PM Kathryn Alexander ***@***.***>
wrote:

> @nalhadef <https://github.com/nalhadef>, this part of the backup and
> restore docs isn't related to the node feature that Michael removed. You
> can open a new PR if you are having trouble fixing this one, but this
> assembly needs to remain. It's been here since 4.5, and it's just an odd
> coincidence that they have similar names.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/65094#issuecomment-1765112542>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT7EEZ755TKH2K4RW2IPDHLX7WAUXAVCNFSM6AAAAAA5DEFUQ6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONRVGEYTENJUGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

lyman9966 commented:
Do I need review another build? Because for https://65094--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-shutdown.html, I can't see anything on "Files changed" tab on the current page

nalhadef commented:
Hi, Min Li.

Yes, please confirm that I addressed your comments properly and didn't misunderstand anything.

Please use this preview: [65094--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-shutdown.html](https://65094--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-shutdown.html)

lyman9966 commented:
Hi, @nalhadef 
I think there are some issues in the preview doc:
1)Example output of "Mark all the nodes in the cluster as unschedulable. You can do this from your cloud provider’s web console, or run the following loop" doesn't align with the previous paragraph
2)some “+” appear on the page, also [source,terminal] appear on the page. I think they shouldn't appear here
3)the serial number of "Evacuate the pods using the following method" should be 3, and "Shut down all of the nodes in the cluster. You can do this from your cloud provider’s web console, or run the following loop" should be 4.(Now they are 1 and 2)
4)there are some irrelevant command like "openssl x509 -in /var/lib/kubelet/pki/kubelet-server-current.pem -noout -enddate"
Indeed, I think the content is a little confusing due to the serial number wronged marked. Please help to check, thanks!

nalhadef commented:
Review sent to QE, QE responded with additional comments.

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -12,99 +12,79 @@ You can shut down your cluster in a graceful manner so that it can be restarted
+node/ci-ln-mgdnf4b-72292-n547t-worker-c-vcmtn cordoned
 ----
-sh-5.1# openssl x509 -in /var/lib/kubelet/pki/kubelet-client-current.pem -noout -enddate
 ----
```
If you remove this line, the build should pass.

kalexand-rh reviewed (commented):
There are a bunch of small format tweaks that are impacting the build. Addressing all of my comments might not make the build pass, but addressing them will make it easier to see any remaining issues.

kalexand-rh reviewed (commented):
@nalhadef, try these tweaks.

> asciidoctor: WARNING: modules/graceful-shutdown.adoc: line 110: unterminated example block

The one at line 110 appears to be correct, but there are two listing block issues earlier in the file.

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -113,17 +95,19 @@ Adjust the shut down time to be longer for large-scale clusters:
 ====
-+
+
+====
```
Remove this `====` and the surrounding blank lines.

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -113,17 +95,19 @@ Adjust the shut down time to be longer for large-scale clusters:
 
 . Shut off any cluster dependencies that are no longer needed, such as external storage or an LDAP server. Be sure to consult your vendor's documentation before doing so.
 +
+
```
Remove this blank line.

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -19,92 +18,74 @@ You can shut down a cluster until a year from the installation date and expect i
 ----
 
-... Change your root directory to `/host` by running the following command:
+<1> To ensure that the cluster can restart gracefully, plan to restart it on or before the specified date. As the cluster restarts, the process might require you to manually approve the pending certificate signing requests (CSRs) to recover kubelet certificates.
```
Remove the blank line before this annotation. The annotation has to go directly after the codeblock.

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -19,92 +18,74 @@ You can shut down a cluster until a year from the installation date and expect i
 
-... Change your root directory to `/host` by running the following command:
+<1> To ensure that the cluster can restart gracefully, plan to restart it on or before the specified date. As the cluster restarts, the process might require you to manually approve the pending certificate signing requests (CSRs) to recover kubelet certificates.
 +
```
Swap the + for a blank line. You need a blank line between steps.

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -12,99 +12,78 @@ You can shut down your cluster in a graceful manner so that it can be restarted
 ----
-<1> `-h 1` indicates how long, in minutes, this process lasts before the control-plane nodes are shut down. For large-scale clusters with 10 nodes or more, set to 10 minutes or longer to make sure all the compute nodes have time to shut down first.
-+
 .Example output
```
You need to add this + to connect the example output to the step and preceding command.

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -19,92 +18,74 @@ You can shut down a cluster until a year from the installation date and expect i
-+
 .Example output
+[source,terminal]
+
```
Remove the blank line here. There's no space between `[source,terminal]` and the codeblock.

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -19,92 +18,74 @@ You can shut down a cluster until a year from the installation date and expect i
 Shutdown scheduled for Mon 2021-09-13 09:36:29 UTC, use 'shutdown -c' to cancel.
 ----
-+
+
```
Restore this +. You need it to chain the output to the following line.

kalexand-rh commented on modules/graceful-shutdown.adoc:
```diff
@@ -113,17 +94,14 @@ Adjust the shut down time to be longer for large-scale clusters:
 ====
 
 . Shut off any cluster dependencies that are no longer needed, such as external storage or an LDAP server. Be sure to consult your vendor's documentation before doing so.
-+
```
You need this + sign.

kalexand-rh commented:
I've opened the `enterprise-4.15` branch. Please review your PR and update the applicable versions to also cover 4.15 if you expect that the changes in this PR will be valid in that branch.

nalhadef commented:
Hi, [Min Li](@lyman9966).

Please take another look at these changes. Thanks!! ([review text](https://65094--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-shutdown.html))

Neal

lyman9966 commented:
/lgtm

nalhadef commented:
/label peer-review-needed

QE (lyman9966) approved.
Can be merged after peer review approval

nalhadef commented:
/label merge-review-needed

michaelryanpeter commented:
/cherrypick enterprise-4.15

michaelryanpeter commented:
/cherrypick enterprise-4.14

michaelryanpeter commented:
/cherrypick enterprise-4.13

bergerhoffer commented:
HI @lyman9966 - We have a request here: https://issues.redhat.com/browse/OSDOCS-9558

It's asking that the graceful shutdown steps to cordon and drain the nodes be backported to 4.12. Do you know whether the changes in this PR could also be backported to 4.12?

lyman9966 commented:
Yeah, these steps can be backported to 4.12

bergerhoffer commented:
/cherrypick enterprise-4.12
